ocaml/xapi/repository_helpers.ml

```diff
@@ -104,14 +104,9 @@ module Pkg = struct
      *   "qemu-dp-2:2.12.0-2.0.11.x86_64", or
      *   "time-(none):1.7-45.el7.x86_64".
      * These may come from "yum updateinfo list updates" and "rpm -qa" *)
-    let r = Re.Str.regexp "^\\(.+\\)\\.\\(noarch\\|x86_64\\)$" in
-    match Re.Str.string_match r s 0 with
-    | true -> (
+    match Astring.String.cut ~rev:true ~sep:"." s with
```

Comment:
Does this work as expected? A package can have more dots and I would expect the result to have a variable list of segments that need to be matched here. The original regexp was anchored at the end and this is why it worked.

Reply:
Yep: `~rev:true` makes the cut happen at the the rightmost dot of the string